### PR TITLE
(PUP-5745) Use PuppetDB 4 API for querying where applicable

### PR DESCRIPTION
**Pull request by nicklewis**
Previously, the CapabilityFinder was using the
Puppetdb::Util::Http.action(...) method to query PuppetDB. In PuppetDB 4
however, the action() method has changed to require a new optional
argument, which breaks compatibility. At the same time, a top-level
query_puppetdb() method has been introduced which is intended to be the
public API going forward. We now check for and use the query_puppetdb()
method where it exists, falling back to the old internal action() method
when necessary.

### CODE REVIEW

**Comment by nicklewis** [comment]
Pending potential further changes in puppetlabs/puppetdb:1824...

**Comment by nicklewis** [comment]
This has been updated per puppetlabs/puppetdb#1824 and should be ready to go.

File: lib/puppet/resource/capability_finder.rb [hunk 1]
```diff
@@ -35,15 +35,10 @@ def self.find(environment, code_id, cap)
           ['select_catalogs',
             ['=', 'code_id', code_id]]]]
     end
-    query = query_terms.to_json
 
-    Puppet.notice "Capability lookup #{cap}]: #{query}"
+    Puppet.notice "Capability lookup #{cap}]: #{query_terms}"
```
Comment (briancain) [inline lib/puppet/resource/capability_finder.rb]: I'm assuming this no longer needs to be json anymore?

Comment (nicklewis) [inline lib/puppet/resource/capability_finder.rb]: It looks basically the same either way (except for spaces), so it seemed like there was no need to JSONify it just for consistency with the old behavior. Could always change it if need be.

Before:
```
["and",["=","type","Sql"],["=","title","foo"],["=","tag","producer:production"],["in","certname",["extract","certname",["select_catalogs",["=","code_id","123456"]]]]]
```

After:
```
["and", ["=", "type", "Sql"], ["=", "title", "foo"], ["=", "tag", "producer:production"], ["in", "certname", ["extract", "certname", ["select_catalogs", ["=", "code_id", "123456"]]]]]
```

Comment (briancain) [inline lib/puppet/resource/capability_finder.rb]: Cool cool!

**Comment by briancain** [comment]
Looks good to me!

**Comment by nicklewis** [comment]
@kylog Ping for visibility as this is blocking orchestrator integration tests. Would be extremely helpful if someone could take a look.

**Comment by kylog** [comment]
:+1:

**Pull request merged**